profiles/permissions.py

```diff
@@ -26,12 +26,20 @@ def has_permission(self, request, view):
         return request.user == view._profile.person
 
 
-class IsOwnCompany(BasePermission):
+class OnlyAdminRead(BasePermission):
     def has_permission(self, request, view):
         user = request.user
-        pk = request.data.get("company_pk")
-        try:
-            profile = Profile.objects.get(id=pk)
-        except ObjectDoesNotExist:
+        if request.method in SAFE_METHODS:
+            return user.is_staff
+        return True
+
+
+class IsOwnCompany(BasePermission):
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
             return True
```

Comment:
I'm approving this PR. But if would be really nice to follow SRP. Ideally, permission should check ONE thing. The class is called `IsOwnCompany` — so it shouldn't care about request method, it should only check if the company is linked to the current user.

And then these permission classes could be combined in the way we need in `permission_classes` view property.